### PR TITLE
Injection webhook: support configure ignored namespaces in helm instead of fixed list.

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**

### CODE REVIEW

**Comment by howardjohn** [comment]
Can you help explain the use case here

File: manifests/charts/istio-control/istio-discovery/values.yaml [hunk 1]
```diff
@@ -137,6 +137,14 @@ sidecarInjectorWebhook:
   #
   # defaultTemplates: ["sidecar", "hello"]
   defaultTemplates: []
+
+  # defaultExcludedNamespaces specifies a set of namespaces that sidecar injection will be skipped.
```
Comment (jacob-delgado) [inline manifests/charts/istio-control/istio-discovery/values.yaml:141]: Should this also extend to not pushing out the `istio-ca-root-cert` configmap to these namespaces as well?

**Comment by linsun** [comment]
Hi @hanxiaop what if it is configured as ignored namespace in helm but user label the namespace as istio injection?

**Comment by hanxiaop** [comment]
> Can you help explain the use case here

@howardjohn The intention here is to provide a more centralized and consistent method for disabling certain namespaces, rather than labeling them separately in complex scenarios, such as multi-cluster environments with global automatic sidecar injection enabled for users. I think this approach also improves flexibility within CI/CD pipeline setups.

However, one concern is whether making this option an add-on to the existing ignored namespaces is better, or if it should override the ignored namespaces values. There may be security issues if the current ignored namespaces are removed in Helm?

**Comment by hanxiaop** [comment]
> Hi @hanxiaop what if it is configured as ignored namespace in helm but user label the namespace as istio injection?

Pods in the ignored namespace will not be injected. If the pod is originally injected with either `sidecar.istio.io/inject=true` or the namespace label `istio-injection=enabled`,and is configured with this new helm option, once a rollout restart happens, the sidecar will be removed.

File: manifests/charts/istio-control/istio-discovery/values.yaml (hunk 1 above)
Comment (hanxiaop) [reply manifests/charts/istio-control/istio-discovery/values.yaml:141]: Good point, I'll update this if the PR is reasonable to be merged. Also, I realized there may be more places that need to be revised.

**Comment by howardjohn** [comment]
IMO the ignored namespaces was a hack because `labelSelector` didn't exist. Its not the right API.

**Pull request closed without merging**